Swiftfin/Views/SettingsView/UserDashboardView/ActiveSessionsView/Components/ActiveSessionRow.swift

```diff
@@ -38,29 +35,47 @@ extension ActiveSessionsView {
         private var rowLeading: some View {
             // TODO: better handling for different poster types
             Group {
-                if session.nowPlayingItem?.type == .audio {
+                switch session.nowPlayingItem {
+                case .none:
                     ZStack {
-                        Color.clear
-
-                        ImageView(session.nowPlayingItem?.squareImageSources(maxWidth: 60) ?? [])
-                            .failure {
-                                SystemImageContentView(systemName: session.nowPlayingItem?.systemImage)
-                            }
-                    }
-                    .squarePosterStyle()
-                } else {
-                    ZStack {
-                        Color.clear
-
-                        ImageView(session.nowPlayingItem?.portraitImageSources(maxWidth: 60) ?? [])
-                            .failure {
-                                SystemImageContentView(systemName: session.nowPlayingItem?.systemImage)
-                            }
+                        DeviceType(
```

Comment:
Instead, we should have a new `device` property on `SessionInfo` that returns the device for that session.

```swift
extension SessionInfo {
  var device: DeviceType {
    // ...
  }
}
```